docs/aws/audit/cloudfrontmonitoring/rules/cloudfront_custom_ssl_certificate.mdx

```diff
@@ -23,6 +23,234 @@ CBP
+
+2. Once the AWS CLI is set up, you can list all the CloudFront distributions using the following command:
+
+   ```
+   aws cloudfront list-distributions --query 'DistributionList.Items[*].{ID:Id,Domain:DomainName}'
+   ```
+   This command will return a list of all the CloudFront distributions along with their ID and domain name.
+
+3. To check the SSL certificate of a specific CloudFront distribution, you can use the following command:
+
+   ```
+   aws cloudfront get-distribution-config --id <distribution-id>
+   ```
+   Replace `<distribution-id>` with the ID of the distribution you want to check. This command will return the configuration of the specified CloudFront distribution.
+
+4. In the output of the above command, look for the `ViewerCertificate` field. If the `ACMCertificateArn` field is present and has a value, it means that a custom SSL certificate is set for the distribution. If the `ACMCertificateArn` field is not present or does not have a value, it means that a custom SSL certificate is not set for the distribution.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3): Before you can start writing Python scripts to manage AWS resources, you need to set up authentication credentials for your AWS account. You can do this by installing AWS CLI and configuring it with your access key ID and secret access key.
+
+   ```bash
+   pip install awscli
+   aws configure
+   ```
+
+2. Import the necessary modules and create a session: In your Python script, you need to import the Boto3 module and create a session using your credentials.
+
+   ```python
+   import boto3
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='us-west-2'
+   )
+   ```
+
+3. List all CloudFront distributions and check for custom SSL certificates: You can use the `list_distributions` method to get a list of all CloudFront distributions. Then, for each distribution, you can check if a custom SSL certificate is set by checking the `ViewerCertificate` attribute.
+
+   ```python
+   client = session.client('cloudfront')
+   distributions = client.list_distributions()
+   for distribution in distributions['DistributionList']['Items']:
+       if 'ViewerCertificate' in distribution and 'ACMCertificateArn' in distribution['ViewerCertificate']:
+           print(f"Custom SSL certificate is set for distribution {distribution['Id']}")
+       else:
+           print(f"Custom SSL certificate is not set for distribution {distribution['Id']}")
+   ```
+
+4. Run the script: Finally, you can run the script to check if a custom SSL certificate is set for all your CloudFront distributions. If a distribution does not have a custom SSL certificate, the script will print a message indicating this.
+
+   ```bash
+   python check_ssl.py
+   ```
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>